pkg/runtime/reconciler.go

```
@@ -360,13 +360,90 @@ func (r *resourceReconciler) Sync(
 	isAdopted := IsAdopted(desired)
 	rlog.WithValues("is_adopted", isAdopted)
 
+	if r.cfg.FeatureGates.IsEnabled(featuregate.ForcedAdoptResources) {
+		
+		// If the resource is being adopted by force, we need to access
+		// the required field passed by annotation and attempt a read.
+		if NeedAdoption(desired) {
+			rlog.WithValues("is_forced_adoption", "true")
+			rlog.Info("Adopting Resource")
+			extractedFields, err := ExtractAdoptionFields(desired)
```

To me this sounds like a user input issue, shall we return a terminal error?